src/NUnitEngine/nunit.engine/Services/RuntimeLocators/NetCoreRuntimeLocator.cs

```diff
@@ -15,100 +13,14 @@ public static IEnumerable<RuntimeFramework> FindRuntimes(bool x86)
         {
             List<Version> alreadyFound = new List<Version>();
 
-            foreach (string dirName in GetRuntimeDirectories(x86))
+            foreach (var runtime in DotNet.GetRuntimes("Microsoft.NETCore.App"))
```

Comment:
This seems to be doing partly what `TestAssemblyResolver` does, but for a different runtime category.
This one doesn't consider the WindowsDektop or AspNetCore runtimes.

Reply:
TestAssemblyResolver is associated with TestAssemblyLoadContext and is used in the agent. It isn't known to the engine at all.

The RuntimeLocators are specific to the engine and, as you say, is used to select agents. Agents (at least current as well as any I am thinking of) are not specific to WindowsDesktop or AspNetCore but only to the specific runtime target. 

Essentially, the engine needs less info to select an agent. The agent then needs a bit more info to correctly load the tests.

Reply:
Thanks, but if the engine instantiated an x86 agent, the TestAssemblyResolver should only look at x86 versions.
That decision should not be up to the callers, but determined by the current runtime using `System.Environment.Is64BitProcess`